src/Middleware/HeadersGenerator.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Implements the HeadersGenerator middleware class
+ *
+ * PHP Version 7
+ *
+ * @category Router
+ * @package  Middleware
+ * @author   Xavier Lecours Boucher <xavier.lecoursboucher@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\Middleware;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \Psr\Http\Server\MiddlewareInterface;
+use \Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * A HeadersGenerator middleware is a chainable type of PSR15-based middleware
+ * which takes a response, and adds an HTTP headers.
+ *
+ * It should be called near the top of a middleware chain (after the response
+ * is fully created)
```

Comment:
I can't decide if this belongs in a middleware or not (maybe check what other frameworks do?, but I think it should be called a HeaderSender if it is, and it should document that it has a side effect of sending headers over the wire rather than modifying the PSR response